scala_proto/scala_proto.bzl

```diff
@@ -319,38 +319,39 @@ def _colon_paths(data):
     return ':'.join(["{root},{path}".format(root=_root_path(f), path=f.path) for f in data])
 
 def _gen_proto_srcjar_impl(ctx):
-    acc_imports = depset()
+    acc_imports = []
     transitive_proto_paths = depset()
 
     proto_deps, jvm_deps = [], []
     for target in ctx.attr.deps:
         if hasattr(target, 'proto'):
             proto_deps.append(target)
-            acc_imports += target.proto.transitive_sources
+            acc_imports.append(target.proto.transitive_sources)
```

Comment:
Isn’t transitive_sources a depset? If so isn’t it better to accumulate them in depsets?

Reply:
there is no way to accumulate depsets that is not deprecated. You accumulate a list and then make a `depset(transitive = your_list_here)`

Reply:
@laurentlb is this correct? Why can’t we accumulate depsets? Turning depsets to lists so we accumulate them as depsets sounds counter intuitive

Reply:
Yes, it's correct (https://docs.bazel.build/versions/master/skylark/depsets.html).
* depset is immutable, so the `+=` operator was misleading
* we can create only one `depset`, with the correct structure, and it's much better for performance (adding depset inside a loop could lead to `O(n^2)` complexity
* it's more explicit: you see that the depset has n transitive depsets at the same level (which corresponds to the internal representation)

Depsets are not turned into lists. We just collect all the depsets first, and then merge them all at the same time.

Reply:
Can we pass a depset to the “transitive” argument of the constructor?
What Oscar did is convert them to lists for this ctor which sounds wrong

Reply:
I didn’t convert them to lists. I added them to a list. The argument to transitive is a list of Depsets. I constructed that list so I could pass the list of depsets to transitive.

Reply:
I see, my bad.
Thanks for the clarification